report_generate_helper/models/report_helper.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# @author Matthieu SAISON <matthieu.saison@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.tools.safe_eval import safe_eval, time
+
+
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    def _get_report_converter(self):
+        return f"_render_{self.report_type.replace('-', '_')}"
+
+    def get_report(self, report_name):
```

Comment:
A better name could be `generate_report`.... `get_report` could mean 'give me the report record for the given report name'

Reply:
Is it intended that this method is public (xml-rpc callable?).

Reply:
No it's not necessary, I change to _generate_report